legal-api/src/legal_api/services/filings/validations/incorporation_application.py

```diff
@@ -186,3 +192,40 @@ def validate_share_structure(incorporation_json) -> Error:
         return msg
 
     return None
+
+
+def validate_incorporation_effective_date(incorporation_json) -> Error:
+    """Return an error or warning message based on the effective date validation rules.
+
+    Rules:
+        - The effective date must be the correct format.
+        - The effective date must be a minimum of 2 minutes in the future.
+        - The effective date must be a maximum of 10 days in the future.
+    """
+    # Setup
+    msg = []
+    now = dt.utcnow()
+    now_plus_2_minutes = now + timedelta(minutes=2)
+    now_plus_10_days = now + timedelta(days=10)
+
+    try:
+        filing_effective_date = incorporation_json['filing']['header']['effectiveDate']
```

Comment:
Are we passing effective date in case of immediate filing? Is the validation passing for immediate filing

Reply:
Also I think the legal API returns the effective date in the filing header when filing is resumed ( after save). It would be a good check to see whether these validations are passing in the second submit

Reply:
So, correct me if i'm wrong, but if it is immediate, effective date is being passed as `null` or `None` to the backend. This check should return the validations with no errors if that is the case. It is the second test scenario. ( At least those were my intentions :) )

Reply:
oh interesting! How could i go about testing that..

Reply:
Try 'save and resume' for an immediate filing. In the second submission(dont change anything just click on file and pay ) if it passes, it is fine

Reply:
kk, will do thanks!